src/lib/FixedLatencyMemoryInterface.cc

```diff
@@ -20,34 +20,35 @@ void FixedLatencyMemoryInterface::tick() {
     }
 
     const auto& target = request.target;
+    uint64_t requestId = request.requestId;
 
     if (request.write) {
-      // Write: write data directly to memory
-      assert(target.address + target.size <= memory_->getMemorySize() &&
-             "Attempted to write beyond memory limit");
-
-      simeng::memory::WriteRespPacket* resp =
-          (simeng::memory::WriteRespPacket*)memory_->requestAccess(
-              new simeng::memory::WritePacket{
-                  target.address, target.size,
-                  request.data.getAsVector<char>()});
-      delete resp;
+      const char* wdata = request.data.getAsVector<char>();
+      std::vector<char> dt(target.size, '\0');
+      std::copy(wdata, wdata + target.size, dt.data());
```

Comment:
To eliminate the copy, couldn't we do `std::vector<char> dt(wdata, wdata + target.size);`?

Reply:
done